x-pack/plugins/task_manager/server/removed_tasks/mark_removed_tasks_as_unrecognized.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/logging';
+import { CoreStart } from '@kbn/core-lifecycle-server';
+import { TaskScheduling } from '../task_scheduling';
+import { TaskTypeDictionary } from '../task_type_dictionary';
+import { ConcreteTaskInstance, TaskManagerStartContract } from '..';
+import { TaskStatus } from '../task';
+import { REMOVED_TYPES } from '../task_type_dictionary';
+import { TASK_MANAGER_INDEX } from '../constants';
+
+export const TASK_ID = 'mark_removed_tasks_as_unrecognized';
+const TASK_TYPE = `task_manager:${TASK_ID}`;
+
+export const SCHEDULE_INTERVAL = '1h';
+
+export async function scheduleMarkRemovedTasksAsUnrecognizedDefinition(
+  logger: Logger,
+  taskScheduling: TaskScheduling
+) {
+  try {
+    await taskScheduling.ensureScheduled({
+      id: TASK_ID,
+      taskType: TASK_TYPE,
+      schedule: { interval: SCHEDULE_INTERVAL },
+      state: {},
+      params: {},
+    });
+  } catch (e) {
+    logger.error(`Error scheduling ${TASK_ID} task, received ${e.message}`);
+  }
+}
+
+export function registerMarkRemovedTasksAsUnrecognizedDefinition(
+  logger: Logger,
+  coreStartServices: () => Promise<[CoreStart, TaskManagerStartContract, unknown]>,
+  taskTypeDictionary: TaskTypeDictionary
+) {
+  taskTypeDictionary.registerTaskDefinitions({
+    [TASK_TYPE]: {
+      title: 'Mark removed tasks as unrecognized',
+      createTaskRunner: taskRunner(logger, coreStartServices),
+    },
+  });
+}
+
+export function taskRunner(
+  logger: Logger,
+  coreStartServices: () => Promise<[CoreStart, TaskManagerStartContract, unknown]>
+) {
+  return () => {
+    return {
+      async run() {
+        try {
```

Comment:
I'd like to have the body of the `try` moved to a local function, to make this easier to read.

Reply:
Done in https://github.com/elastic/kibana/pull/199057/commits/9bb08996e96f3a66e1bb009ea7fd46b21bc0cadd